app/controllers/api/correct_answers_controller.rb

```diff
@@ -8,6 +8,7 @@ def create
     @answer = @question.answers.find(params[:answer_id])
     @answer.type = 'CorrectAnswer'
     if @answer.save
+      Newspaper.publish(:answer_destroy, @answer)
```

Comment:
ベストアンサーの有無による解決質問数の変化については`app/controllers/api/answers_controller.rb`の`update`でキャッシュ削除処理を制御できているんじゃないかな…？と思ったのですがいかがでしょうか…？🤔ここに追記されている理由がありましたらご教授いただけると嬉しいな…と思いました…🙏22行目についても同様です。

Reply:
@choco0809 
すいません、わたしの上記の指摘ではキャッシュ削除処理は制御できないな〜と改めて手元で確認できたので、この指摘はなしとさせてください〜🙏ごめんなさい。！
```suggestion
      Newspaper.publish(:answer_save, @answer)
```
とした方がいいかな〜と思いました。`update`の箇所も同様です〜☺️

Reply:
@pachikuriii 
了解しました～
ちなみにキャッシュ削除処理が制御できているかどうやって確認されましたか？？？
自分も気になって`Rails.Cache`で確認しようとしているのですがイマイチぴんとこなくて...

Reply:
@choco0809 

お返事ありがとうございます〜！
chocoさんがPRの確認方法の中で紹介されていた方法で、回答とベストアンサーを操作したときに`call`メソッドがきちんと呼ばれる＝キャッシュ削除処理が制御できていると捉えていました。

`Rails.Cache`で確認しようとされているということは、キャッシュ削除そのものができているかどうか？の確認のことをおっしゃっているのだと思うんですが、わたしが思いつく限りではその確認においても[app/models/answer_cache_destroyer.rb](https://github.com/fjordllc/bootcamp/pull/5525/files/8830a7fd5a940550f0d7f718c5a4bf737c943436#diff-bd8ebcb155ab232c787cb3d620e956fd669ef65e681325aedb8efe766e5fa012)の`call`メソッドの中で下記のように確認するしか思いつかなかったため、`call`メソッドの中に`puts`を挟む方法で確認したらいいのかな〜と考えておりました…🙏！

```ruby
class AnswerCacheDestroyer
  def call(_answer)
    puts Cache.not_solved_question_count # キャッシュ削除前の確認
    Cache.delete_not_solved_question_count # キャッシュ削除
    puts Cache.not_solved_question_count # キャッシュ削除後の確認
  end
end
```

上記のようにコードを書いて実行した時のログは以下です。

**ベストアンサーを付与したとき**
`Cache.not_solved_question_count`で返される解決質問数がきちんと変動するのでキャッシュ削除が行われていると判断していいかなと思いました。
```
 ↳ app/controllers/api/correct_answers_controller.rb:10:in `create'
  TRANSACTION (41.9ms)  COMMIT
  ↳ app/controllers/api/correct_answers_controller.rb:10:in `create'
40
# ここでキャッシュ削除の処理 Cache.delete_not_solved_question_count
   (3.5ms)  SELECT COUNT(*) FROM "questions" WHERE "questions"."id" NOT IN (SELECT "answers"."question_id" FROM "answers" WHERE "answers"."type" = $1)  [["type", "CorrectAnswer"]]
  ↳ app/models/cache.rb:47:in `block in not_solved_question_count'
39
```

参考までに`Cache.delete_not_solved_question_count`を無くすと、ベストアンサーを付与しても`Cache.not_solved_question_count`で返される解決質問数は変動しません。キャッシュが残って解決質問数に反映されないということなのかなーと思いました。
```
   ↳ app/controllers/api/correct_answers_controller.rb:10:in `create'
  TRANSACTION (14.7ms)  COMMIT
  ↳ app/controllers/api/correct_answers_controller.rb:10:in `create'
40
# キャッシュ削除の処理 Cache.delete_not_solved_question_countはなし
40
```

なので、`call`メソッドの中の`Cache.delete_not_solved_question_count`がきちんと呼ばれるか？が肝かな…と自分なりに考えていました…どうなん？なポイントありましたらツッコミいただけると嬉しいです！＆他にいい方法があれば、知りたいなーと思いました✨

Reply:
@pachikuriii 
ご丁寧にありがとうございます🙇‍♂️🙇‍♂️🙇‍♂️
自分もキャッシュの確認方法について調べてみたのですが、あまりピンとくるものがなく...
結局、pachikuriiiさんと同じ手法でキャッシュが削除されているかの確認しか思いつきませんでした！！！